File: content/snippets/python/interview_lcci/16/21_sum_swap.py
from typing import List


def find_swap_values(array1: List[int], array2: List[int]) -> List[int]:
    diff = sum(array1) - sum(array2)
    if diff % 2 != 0:
        return []
    target = diff // 2
    set2 = set(array2)
    for x in array1:
        y = x - target
        if y in set2:
            return [x, y]
    return []


def main() -> None:
    print(find_swap_values([4, 1, 2, 1, 1, 2], [3, 6, 3, 3]))


if __name__ == "__main__":
    main()

File: content/snippets/scala2/interview_lcci/16/21_sum_swap.scala
object Lcci1621SumSwap {
    def findSwapValues(array1: Array[Int], array2: Array[Int]): Array[Int] = {
        val diff = array1.sum - array2.sum
        if (diff % 2 != 0) return Array.emptyIntArray
        val target = diff / 2
        val set2 = array2.toSet
        array1.foreach { x =>
            val y = x - target
            if (set2.contains(y)) return Array(x, y)
        }
        Array.emptyIntArray
    }

    def main(args: Array[String]): Unit = {
        println(findSwapValues(Array(4, 1, 2, 1, 1, 2), Array(3, 6, 3, 3)).mkString("[", ", ", "]"))
    }
}
